.github/workflows/release.yml

```diff
@@ -0,0 +1,52 @@
+name: Release to crates.io
+
+on:
+  push:
+    tags:
+      - "v*.*.*" # Trigger on version tags like v0.1.9
+
+env:
+  CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: main
+          fetch-depth: 0
+
+      - name: Set up Rust toolchain
+        uses: dtolnay/rust-toolchain@stable # Ensure the stable Rust version
+
+      - name: Verify Cargo Credentials
+        env:
+          CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
+        run: |
+          if [ -z "$CARGO_REGISTRY_TOKEN" ]; then
+            echo "Error: CARGO_REGISTRY_TOKEN is not set."
+            exit 1
+          fi
+
+      - name: Publish to crates.io
+        run: cargo publish --token $CARGO_REGISTRY_TOKEN
+
+      - name: Commit and Tag the Release
```

Comment:
I don't understand this part. This workflow is triggered when a tag is added. What tag are we creating here?

Reply:
This step helps to understand the published part locally. It will be removed in the final go.

Reply:
removed the tag release step